frontend/src/components/QuickAddCartButton/QuickAddCartButton.tsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  MinusOutlined, PlusOutlined,
+} from '@ant-design/icons';
+import { Button, Tooltip } from 'antd';
+import axios from 'axios';
+import { Course, UnselectCourses } from 'types/api';
+import { PlannerCourse } from 'types/planner';
+import prepareUserPayload from 'utils/prepareUserPayload';
+import type { RootState } from 'config/store';
+import { addToUnplanned, removeCourses } from 'reducers/plannerSlice';
+import S from './styles';
+
+type Props = {
+  courseCode: string
+  planned?: boolean
+};
+
+const QuickAddCartButton = ({ courseCode, planned }: Props) => {
+  const dispatch = useDispatch();
+
+  const { degree, planner } = useSelector((state: RootState) => state);
+
+  const addToPlanner = async (e: React.MouseEvent<HTMLElement>, code: string) => {
+    e.stopPropagation();
+    try {
+      const { data: course } = await axios.get<Course>(`/courses/getCourse/${code}`);
+
+      const courseData: PlannerCourse = {
```

Comment:
use {...course}
Also, i swear this functionality is somewhere else too, this should be a function somewhere